lib/sockutils.c

```diff
@@ -345,7 +345,12 @@ gftp_fd_set_sockblocking (gftp_request * request, int fd, int non_blocking)
   else
     flags &= ~O_NONBLOCK;
 
+#ifdef __APPLE__
+  int opt = 1;
```

Comment:
This will only turn on non-blocking sockets but it will not turn it back off. This should be: `int opt = non_blocking;` since the non-blocking mode is switched in the code base.

Does this work for you on OS/X? I don't have access to that platform so I can't test this functionality there. O_NONBLOCK is defined by POSIX and is the preferred way of doing that.

More of this code can be wrapped in an #ifdef block. Look further up inside that same function. The existing socket flags and fetched, modified, and then the socket is set to non-blocking. So you should be able to do:

$ifdef __APPLE__
     int opt = non_blocking;
     if (ioctl(fd, FIONBIO, &opt) < 0) 
#else
     // Most of the existing function...